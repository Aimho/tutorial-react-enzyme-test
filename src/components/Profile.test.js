import React from "react";
import { mount } from "enzyme";
import Profile from "./Profile";

describe("<Profile />", () => {
  it("matches snapshot", () => {
    const wrapper = mount(<Profile userName="AimHo" name="조준호" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders userName and name", () => {
    const wrapper = mount(<Profile userName="AimHo" name="조준호" />);
    expect(wrapper.props().userName).toBe("AimHo");
    expect(wrapper.props().name).toBe("조준호");

    const boldElement = wrapper.find("b");
    expect(boldElement.contains("AimHo")).toBe(true);
    const spanElement = wrapper.find("span");
    expect(spanElement.text()).toBe("조준호");
  });
});
