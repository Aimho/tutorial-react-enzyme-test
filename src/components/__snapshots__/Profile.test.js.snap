// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<Profile
  name="조준호"
  userName="AimHo"
>
  <div>
    <b>
      AimHo
      !
    </b>
    <span>
      조준호
    </span>
  </div>
</Profile>
`;
